internal/provider/resource_application_test.go

```diff
@@ -137,6 +137,38 @@ func TestAcc_ResourceApplication_Updates(t *testing.T) {
 	})
 }
 
+// TestAcc_ResourceApplication_UpdatesRevisionConfig will test the revision update that have new config parameters on
+// the charm. The test will check that the config is updated and the revision is updated as well.
+func TestAcc_ResourceApplication_UpdatesRevisionConfig(t *testing.T) {
+	modelName := acctest.RandomWithPrefix("tf-test-application")
+	appName := "github-runner"
+	resource.Test(t, resource.TestCase{
+		PreCheck:                 func() { testAccPreCheck(t) },
+		ProtoV6ProviderFactories: frameworkProviderFactories,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccResourceApplicationWithRevisionAndConfig(modelName, appName, 8, ""),
+				Check: resource.ComposeTestCheckFunc(
+					resource.TestCheckResourceAttr("juju_application.this", "model", modelName),
+					resource.TestCheckResourceAttr("juju_application.this", "charm.#", "1"),
+					resource.TestCheckResourceAttr("juju_application.this", "charm.0.name", appName),
+					resource.TestCheckResourceAttr("juju_application.this", "charm.0.revision", "8"),
```

Comment:
You're referencing an application resource named "this" which doesn't exist in your template. The resource is named "hithub-runner"